File: Pelikula.API/Model/Korisnik/KorisnikResponse.cs
using Pelikula.API.Model.TipKorisnika;
using System;

namespace Pelikula.API.Model.Korisnik
{
    public partial class KorisnikResponse
    {
        public int Id { get; set; }
        public string KorisnickoIme { get; set; }
        public string Ime { get; set; }
        public string Prezime { get; set; }
        public string Email { get; set; }
        public string Spol { get; set; }
        public DateTime? DatumRodjenja { get; set; }
        public byte[] Slika { get; set; }
        public byte[] SlikaThumb { get; set; }
        public string LozinkaHash { get; set; }
        public string LozinkaSalt { get; set; }
        public string Lozinka { get; set; }

        public TipKorisnikaResponse TipKorisnika { get; set; }
    }
}

File: Pelikula.MOBILE/lib/model/korisnik/korisnik_response.dart
import 'dart:convert';

import 'package:pelikula_mobile/model/korisnik/tip_korisnika_response.dart';

class KorisnikResponse {
  int? id;
  String? korisnickoIme;
  String? ime;
  String? prezime;
  String? email;
  String? spol;
  DateTime? datumRodjenja;
  String? lozinkaHash;
  String? lozinkaSalt;
  String? lozinka;

  TipKorisnikaResponse? tipKorisnika;

  KorisnikResponse(
      {this.id,
      this.korisnickoIme,
      this.ime,
      this.prezime,
      this.email,
      this.spol,
      this.datumRodjenja,
      this.lozinkaHash,
      this.lozinkaSalt,
      this.lozinka,
      this.tipKorisnika});

  factory KorisnikResponse.fromJson(Map<String, dynamic> json) {
    return KorisnikResponse(
        id: json['id'] as int,
        korisnickoIme: json['korisnickoIme'] as String,
        ime: json['ime'] as String,
        prezime: json['prezime'] as String,
        email: json['email'] as String,
        spol: json['spol'] as String,
        datumRodjenja: DateTime.tryParse(json['datumRodjenja']),
        lozinkaHash: json['lozinkaHash'] as String,
        lozinkaSalt: json['lozinkaSalt'] as String,
        lozinka: json['lozinka'] != null ? json['lozinka'] as String : null,
        tipKorisnika: TipKorisnikaResponse.fromJson(json['tipKorisnika']));
  }

  Map<String, dynamic> toJson() => {
        "id": id,
        "korisnickoIme": korisnickoIme,
        "ime": ime,
        "prezime": prezime,
        "email": email,
        "spol": spol,
        "datumRodjenja":
            datumRodjenja == null ? null : datumRodjenja!.toIso8601String(),
        "lozinkaHash": lozinkaHash,
        "lozinkaSalt": lozinkaSalt,
        "lozinka": lozinka,
        "tipKorisnika": tipKorisnika!.toJson(),
      };
}
